packages/node-core/src/indexer/blockDispatcher/base-block-dispatcher.ts

```diff
@@ -181,6 +181,13 @@ export abstract class BaseBlockDispatcher<Q extends IQueue, DS, B> implements IB
           void this.poiSyncService.syncPoi();
         }
         this.eventEmitter.emit(IndexerEvent.RewindSuccess, {success: true, height: reindexBlockHeader.blockHeight});
+
+        // End transaction
+        await this.storeModelProvider.applyPendingChanges(
+          height,
+          !this.projectService.hasDataSourcesAfterHeight(height),
+          this.storeService.transaction
+        );
```

Comment:
Should this happen before the rewind is called? Otherwise it will save data from a block after the rewind?